### DISCUSSION TITLE
Customizable background for low/urgent in Notify widget.

### DISCUSSION THREAD

**Discussion by ftpd** [qtile/qtile#3955] [Ideas]
Currently widget [notify](http://docs.qtile.org/en/stable/manual/ref/widgets.html#notify) allows to set foreground for normal, low and urgent notifications. It would be nice to have a possibility to set background for low and urgent notifications too.

**Comment by elParaguayo** [comment]
You should be able to do it by creating a new widget based on the `Notify` widget. Something like this:
```python
from libqtile.widget import Notify

class NewNotify(Notify):
    defaults = [
        ("background_urgent", "440000", "Background urgent priority colour"),
        ("background_low", "444444", "Background low priority colour"),
    ]

    def __init__(self, width=bar.CALCULATED, **config):
        Notify.__init__(self, width, **config)
        self.add_defaults(NewNotify.defaults)
        self.background_normal = self.background

    def set_notif_text(self, notif):
        Notify.set_notif_text(self, notif)
        urgency = getattr(notif.hints.get("urgency"), "value", 1)
        if urgency == 2:
            self.background = self.background_urgent
        elif urgency == 0:
            self.background = self.background_low
        else:
            self.background = self.background_normal
```

You can then put `NewNotify()` in your bar.

This is untested so may have some bugs!

**Reply by m-col** [reply]
Nice! Perhaps it makes sense to add this to the in-built widget?

**Reply by ftpd** [reply]
@elParaguayo thanks, but I still support @m-col's question: would it be possible to have it built-in?

**Reply by elParaguayo** [reply]
Yes - #3958.

**Reply by ftpd** [reply]
Thank you! I will observe the PR and update as soon as it's merged.
